flytekit/core/base_task.py

```diff
@@ -236,7 +245,17 @@ def _local_execute(self, ctx: FlyteContext, **kwargs) -> Union[Tuple[Promise], P
         )
         input_literal_map = _literal_models.LiteralMap(literals=kwargs)
 
-        outputs_literal_map = self.dispatch_execute(ctx, input_literal_map)
+        # if metadata.cache is set, check memoized version
+        if self._metadata.cache:
+            # The cache key is composed only of '(input_literal_map, cache_version)', i.e. all other parameters
```

Comment:
should we also include the task name as part of the key?

similar to what we do with the identifier on hosted flyte https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/catalog/client.go#L26

Reply:
The key is already implicitly part of the cache key (in other words, `joblib.Memory` only caches the result of _each_ function call). I'm going to update the comment to reflect that.

Reply:
Actually, this is a really good call. I'm going to add a test case to demonstrate the problem and subsequent fix.

Reply:
Fixed by https://github.com/flyteorg/flytekit/pull/592/commits/130b4327c7e9b5bad40626c0bdc52669270d97b9.